spark/src/main/java/io/ddf/spark/content/SchemaHandler.java

```diff
@@ -106,5 +127,260 @@ public static Schema getSchemaFromDataFrame(DataFrame rdd) {
+   *
+   * @param row:          Which row to convert
+   * @param index:        index of column in row to convert
+   * @param requiredType: type we need to convert to
+   * @return whether we could convert row[index] to requiredType
+   */
+  private boolean isConvertible(Row row, Integer index, Schema.ColumnType requiredType) {
+
+    try {
+      if(row.get(index) == null) return false;
+      String columnData = row.get(index).toString().trim();
+      if (columnData == null) return false;
+      switch (requiredType) {
+        case TINYINT:
+          return Byte.valueOf(columnData) != null;
+        case SMALLINT:
+          return Short.valueOf(columnData) != null;
+        case INT:
+          return Integer.valueOf(columnData) != null;
+        case BIGINT:
+          return Long.valueOf(columnData) != null;
+        case FLOAT:
+          return Float.valueOf(columnData) != null;
+        case DOUBLE:
+          return Double.valueOf(columnData) != null;
+        case DECIMAL:
+          return new BigDecimal(columnData) != null;
+        case STRING:
+          return true;
+        case BINARY:
+          return columnData.getBytes() != null;
+        case BOOLEAN:
+          return Boolean.valueOf(columnData) != null;
+        case TIMESTAMP:
+          return Timestamp.valueOf(columnData) != null;
+        case DATE:
+          return Date.valueOf(columnData) != null;
+        case ARRAY:
+        case ANY:
+        case BLOB:
+          return true;
+        case STRUCT:
+        case MAP:
+          return false; //need more info to parse
+      }
+    } catch (ClassCastException | NullPointerException | IllegalArgumentException nfe) {
+      mLog.debug(nfe.getMessage());
+    }
+    return false;
+  }
+
+  /**
+   * get list column need to be convert
+   *
+   * @return List<Tuple3<Integer,String,ColumnType>> = ColumnIndex,ColumnName,ColumnType
+   */
+  protected List<Tuple3<Integer, String, Schema.ColumnType>> getListConvertColumn(Schema fromSchema, Schema toSchema) throws DDFException {
+    List<Tuple3<Integer, String, Schema.ColumnType>> listConvertColumn = new ArrayList<>();
+    Map<String, Integer> mapNameIndex = new HashMap<>();
+    List<String> columnNames = fromSchema.getColumnNames();
+    int i = 0;
+    for (String name : columnNames) {
+      mapNameIndex.put(name, i++);
+    }
+    for (Column column : toSchema.getColumns()) {
+      if (!mapNameIndex.containsKey(column.getName())) throw new DDFException("Unknown column: " + column.getName());
+      listConvertColumn.add(new Tuple3(mapNameIndex.get(column.getName()), column.getName(), column.getType()));
+    }
+    return listConvertColumn;
+  }
+
+  /**
+   * ApplySchemaStatistic will hold: + total line processed + total line success + column failed parsed info (total
+   * failed on this column & example )
+   */
+  @NotThreadSafe
+  public static class ApplySchemaStatistic implements Serializable {
```

Comment:
kk I also don't know :v 
You could check it here: https://docs.google.com/document/d/1dSw-4zPXq5HQ99-VO1Pt0fyb58slRus8wZnlzp90ftI/edit#heading=h.fm8rwtgx02uq